lib/ffi-compiler/compile_task.rb

```diff
@@ -133,6 +133,11 @@ def define_task!
               sh "#{cc} #{cflags} -o #{t.name} -c #{t.prerequisites[0]}"
             end
 
+          elsif src =~ /\.go$/
+            file obj_file => [ src, File.dirname(obj_file) ] do |t|
+              sh "#{go} build -buildmode=c-shared -o #{t.name} #{t.prerequisites[0]}"
```

Comment:
I know that current code already doesn't do that, but we really should escape arguments for shell properly with `Shellwords.escape`. Could even use `Shellwords.join`.

Reply:
Be careful with shellwords - it's [not compatible to Windows cmd or powershell](https://github.com/oneclick/rubyinstaller2/wiki/For-gem-developers#user-content-shell-escaping). It's best to use list or Array form of methods like:
```ruby
sh go, "build", "-buildmode=c-shared", "-o", t.name, t.prerequisites[0]
```